### PR TITLE
fix: navigation crash on expand non-present parent ids

### LINKED ISSUE #572: [bug] Can not navigate when child is open as root
## To reproduce the issue
- create a structure like the following:
  - A (root item)
    - B (first child)
      -  C (child of child)
- navigate to B in the builder
- open B in the player
- navigate to C in the player (using navigation or next-prev-buttons, both with produce the bug)

## Why this happens

It looks like the issue is when opening non root items as player roots.
In fact, when navigating, the tree view (navigation in the drawer) expands all parents of the current item. But when the root player item is not an absolute root, we try to expand items that are not present in the tree and the tree component crashes.

## Solution 

We should filter the item ids to expand to only include the accessible items (the ones that are at or blow the player root).

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR:
- fix #572

### CODE REVIEW

File: src/modules/navigation/tree/TreeView.tsx [hunk 1]
```diff
@@ -100,6 +100,13 @@ const TreeView = ({
     ? getIdsFromPath(focusedItem.path)
     : defaultExpandedIds;
 
+  // need to filter the expandedIds to only include items that are present in the tree
+  // we should not include parents that are above the current player root
+  const availableItemIds = itemsToShow?.map(({ id: itemId }) => itemId);
```
Comment (pyphilia) [inline src/modules/navigation/tree/TreeView.tsx:105]: Why does `itemsToShow` contain "empty" id in the first place? Maybe you want to filter higher in the logic.

File: src/modules/navigation/tree/TreeView.tsx [hunk 2]
```diff
@@ -100,6 +100,13 @@ const TreeView = ({
...
+  const accessibleExpandedItems = expandedIds.filter((e) =>
```
Comment (pyphilia) [inline src/modules/navigation/tree/TreeView.tsx:106]: Maybe you want to check `availableItemIds`'s length so you don't loop on `expandedIds` for nothing?

**Review by pyphilia — APPROVED** [review]
Thanks!!

File: src/modules/navigation/tree/TreeView.tsx (hunk 1 above)
Comment (spaenleh) [reply src/modules/navigation/tree/TreeView.tsx:105]: Not sure I get your question. This line simply extracts the ids of the available items. It is not used for filtering empty items.

**Pull request merged**